sentry-android-core/src/main/java/io/sentry/android/core/internal/tombstone/TombstoneParser.java

```diff
@@ -0,0 +1,219 @@
+        tombstone.getThreadsMap().entrySet()) {
+      final TombstoneProtos.Thread threadEntryValue = threadEntry.getValue();
+
+      final SentryThread thread = new SentryThread();
+      thread.setId(Long.valueOf(threadEntry.getKey()));
+      thread.setName(threadEntryValue.getName());
+
+      final SentryStackTrace stacktrace = createStackTrace(threadEntryValue);
+      thread.setStacktrace(stacktrace);
+      if (tombstone.getTid() == threadEntryValue.getId()) {
+        thread.setCrashed(true);
+        // even though we refer to the thread_id from the exception,
+        // the backend currently requires a stack-trace in exception
+        exc.setStacktrace(stacktrace);
+      }
+      threads.add(thread);
+    }
+
+    return threads;
+  }
+
+  @NonNull
+  private static SentryStackTrace createStackTrace(@NonNull final TombstoneProtos.Thread thread) {
+    final List<SentryStackFrame> frames = new ArrayList<>();
+
+    for (TombstoneProtos.BacktraceFrame frame : thread.getCurrentBacktraceList()) {
+      final SentryStackFrame stackFrame = new SentryStackFrame();
+      stackFrame.setPackage(frame.getFileName());
+      stackFrame.setFunction(frame.getFunctionName());
+      stackFrame.setInstructionAddr(formatHex(frame.getPc()));
+      frames.add(0, stackFrame);
+    }
+
+    final SentryStackTrace stacktrace = new SentryStackTrace();
+    stacktrace.setFrames(frames);
+
+    // `libunwindstack` used for tombstones already applies instruction address adjustment:
+    // https://android.googlesource.com/platform/system/unwinding/+/refs/heads/main/libunwindstack/Regs.cpp#175
+    // prevent "processing" from doing it again.
+    stacktrace.setInstructionAddressAdjustment(SentryStackTrace.InstructionAddressAdjustment.NONE);
+
+    final Map<String, String> registers = new HashMap<>();
+    for (TombstoneProtos.Register register : thread.getRegistersList()) {
+      registers.put(register.getName(), formatHex(register.getU64()));
+    }
+    stacktrace.setRegisters(registers);
+
+    return stacktrace;
+  }
+
+  @NonNull
+  private List<SentryException> createException(@NonNull TombstoneProtos.Tombstone tombstone) {
+    final SentryException exception = new SentryException();
+
+    if (tombstone.hasSignalInfo()) {
+      final TombstoneProtos.Signal signalInfo = tombstone.getSignalInfo();
+      exception.setType(signalInfo.getName());
+      exception.setValue(excTypeValueMap.get(signalInfo.getName()));
+      exception.setMechanism(createMechanismFromSignalInfo(signalInfo));
+    }
+
+    exception.setThreadId((long) tombstone.getTid());
+    final List<SentryException> exceptions = new ArrayList<>(1);
+    exceptions.add(exception);
+
+    return exceptions;
+  }
+
+  @NonNull
+  private static Mechanism createMechanismFromSignalInfo(
+      @NonNull final TombstoneProtos.Signal signalInfo) {
+
+    final Mechanism mechanism = new Mechanism();
+    // this follows the current processing triggers strictly, changing any of these
```

Comment:
I guess grouping is going to be different anyway because of properly symbolicated stacktraces compared to sentry-native? That said, I would be fine with changing the mechanism to "tombstone" now, even if it breaks existing dashboards/filters. We should just be explicit in the changelog perhaps that this requires new filters etc.? But not strong opinion, signalhandler is also fine (+ we can filter by sdk.name if we wanna whether this comes from sentry-native or the tombstone integration)